packages/jest-resolve/src/defaultResolver.ts

```diff
@@ -55,22 +64,76 @@ export default function defaultResolver(
     return pnpResolver(path, options);
   }
 
-  const result = resolveSync(path, {
+  const result = resolveSync(path, getSyncResolveOptions(options));
+
+  // Dereference symlinks to ensure we don't create a separate
+  // module instance depending on how it was referenced.
+  return realpathSync(result);
+}
+
+export async function defaultResolverAsync(
+  path: Config.Path,
+  options: ResolverOptionsAsync,
+): Promise<Config.Path> {
+  // Yarn 2 adds support to `resolve` automatically so the pnpResolver is only
+  // needed for Yarn 1 which implements version 1 of the pnp spec
+  if (process.versions.pnp === '1') {
+    // @ts-expect-error until https://github.com/arcanis/jest-pnp-resolver/pull/10 is released
+    return pnpResolver(path, options);
+  }
+
+  return new Promise((resolve, reject) => {
+    function resolveCb(err: Error | null, result?: string) {
+      if (err) {
+        reject(err);
+      }
+      if (result) {
```

Comment:
this means that if we don't get a result the promise hangs forever. We should resolve with the result regardless of its value